core/types.ml

```diff
@@ -124,6 +124,61 @@ type ('t, 'r) session_type_basis =
     | `End ]
       [@@deriving show]
 
+type lens_delete_mode =
+  [ `DeleteLeft
+  | `DeleteRight
+  | `DeleteBoth ]
+      [@@deriving show]
+
+type lens_phrase = 
```

Comment:
What is the relevance of `lens_phrase` to types? It seems to be some representation of term syntax.

Reply:
The original work on relational lenses defines typing rules which involve the phrases. Not all of these checks are done yet, and most of this depends also on how possible it is to instantiate "dynamic" phrases. Having `lens_phrase` in the type allows some of those static type checks to be performed during compile time but limits functionality. These checks aren't quite implemented yet, but I have been trying to retain the concept of having a `lens_sort` describing known information about lenses.